plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -556,19 +556,26 @@ public synchronized void renameTable(HiveIdentity identity, String databaseName,
         Table table = getRequiredTable(databaseName, tableName);
         getRequiredDatabase(newDatabaseName);
 
-        if (isIcebergTable(table)) {
-            throw new TrinoException(NOT_SUPPORTED, "Rename not supported for Iceberg tables");
-        }
-
         // verify new table does not exist
         verifyTableNotExists(newDatabaseName, newTableName);
 
         Path oldPath = getTableMetadataDirectory(databaseName, tableName);
         Path newPath = getTableMetadataDirectory(newDatabaseName, newTableName);
 
         try {
-            if (!metadataFileSystem.rename(oldPath, newPath)) {
-                throw new TrinoException(HIVE_METASTORE_ERROR, "Could not rename table directory");
+            if (isIcebergTable(table)) {
+                if (!metadataFileSystem.mkdirs(newPath)) {
+                    throw new TrinoException(HIVE_METASTORE_ERROR, "Could not create new table directory");
+                }
+                if (!metadataFileSystem.rename(new Path(oldPath, TRINO_SCHEMA_FILE_NAME), new Path(newPath, TRINO_SCHEMA_FILE_NAME))) {
```

Comment:
can we add a comment saying that we do this for metadata referencing the old path?

Reply:
sure